GoogleUtilities/Environment/GULHeartbeatDateStorage.m

```diff
@@ -95,65 +93,68 @@ + (void)checkAndCreateDirectory:(NSURL *)directoryPathURL
+    heartbeatDictionary = [GULSecureCoding unarchivedObjectOfClasses:objectClasses
+                                                            fromData:objectData
+                                                               error:&error];
+    if (heartbeatDictionary.count == 0 || error != nil) {
+      heartbeatDictionary = [NSDictionary new];
     }
   }
-  return dict;
+
+  return heartbeatDictionary;
 }
 
 - (nullable NSDate *)heartbeatDateForTag:(NSString *)tag {
-  __block NSMutableDictionary *dict;
+  __block NSDictionary *heartbeatDictionary;
   NSError *error;
   [self.fileCoordinator coordinateReadingItemAtURL:self.fileURL
                                            options:0
                                              error:&error
                                         byAccessor:^(NSURL *readingURL) {
-                                          dict = [self heartbeatDictionaryWithFileURL:readingURL];
+                                          heartbeatDictionary =
+                                              [self heartbeatDictionaryWithFileURL:readingURL];
                                         }];
-  return dict[tag];
+  return heartbeatDictionary[tag];
 }
 
 - (BOOL)setHearbeatDate:(NSDate *)date forTag:(NSString *)tag {
   NSError *error;
   __block BOOL isSuccess = false;
-  [self.fileCoordinator coordinateReadingItemAtURL:self.fileURL
-                                           options:0
-                                  writingItemAtURL:self.fileURL
-                                           options:0
-                                             error:&error
-                                        byAccessor:^(NSURL *readingURL, NSURL *writingURL) {
-                                          NSMutableDictionary *dictionary =
-                                              [self heartbeatDictionaryWithFileURL:readingURL];
-                                          dictionary[tag] = date;
-                                          NSError *error;
-                                          isSuccess = [self writeDictionary:dictionary
-                                                              forWritingURL:writingURL
-                                                                      error:&error];
-                                        }];
+  [self.fileCoordinator
+      coordinateReadingItemAtURL:self.fileURL
+                         options:0
+                writingItemAtURL:self.fileURL
+                         options:0
+                           error:&error
+                      byAccessor:^(NSURL *readingURL, NSURL *writingURL) {
+                        NSMutableDictionary *heartbeatDictionary =
+                            [[self heartbeatDictionaryWithFileURL:readingURL] mutableCopy];
+                        heartbeatDictionary[tag] = date;
+                        NSError *error;
+                        isSuccess = [self writeDictionary:heartbeatDictionary.copy
```

Comment:
Rather than passing around a mutable type for the `heartbeatDictionary`, it's been changed so the dictionary is passed around as an immutable dictionary until it is time to update it with a `tag`. Then, a mutable copy is made, updated, and then an immutable copy is sent to be written to file. 

I was thinking that treating the `heartbeatDictionary` more like a value type here could be more safe.

Is it ok that I've changed the type that is written to file?